app/assets/javascripts/avalon.js

```diff
@@ -47,4 +47,6 @@ $(document).ready(function() {
     })
     $('input[readonly], textarea[readonly]').attr("readonly", false);
   }
+
+  $('a').has('img, ul').addClass('block');
```

Comment:
The idea for using JS was to avoid needing to remember to add the class for future  changes, for what is mostly a visual fix. No problem changing it if others prefer it in the HTML.